src/hoc/withInputInfo.tsx

```diff
@@ -0,0 +1,34 @@
+import * as React from 'react';
```

Comment:
You have wrong understanding of High Order Components: High order component stands for the **enhancing** of other basic component. What are you doing here is a conditional rendering, which can be easily archived with a normal component. Please refactor it to `InputNotification` and add to the components folder.

Reply:
`InputNotification` cannot be a react component. 
Just take a look what's returning by `withInputInfo` (and I think it's kinda enhancing).

Reply:
Please read carefully the documentation:
https://reactjs.org/docs/higher-order-components.html

> Concretely, a higher-order component is a function that takes a component and returns a new component.

It's called **enhancing** .

Reply:
Can we consider `React.ReactChild` as a component?

Reply:
React component:
```
const Component = () => <div />
```

ReactChild, which is also a `ReactElement | string | number`:
```
<Component />
```

A good article describing principles:
https://medium.com/@fay_jai/react-elements-vs-react-components-vs-component-backing-instances-14d42729f62

Reply:
```

export interface InputNotificationProps extends InputErrorProps, InputInfoProps {
  error?: React.ReactNode;
  info?: React.ReactNode;
}

export const InputNotification: React.FC<InputNotificationProps> = ({ error, info, ...rest }) => {
  if (error) {
    if (typeof error === 'string') {
      return <InputError {...rest}>{error}</InputError>;
    }

    return <>{error}</>;
  } else if (info) {
    if (typeof info === 'string') {
      return <InputInfo {...rest}>{info}</InputInfo>;
    }

    return <>{info}</>;
  }

  // tslint:disable-next-line
  return null;
};

/**
 * @deprecated use InputNotification component instead
 * @param error ReactChild
 * @param info ReactChild
 */
export function showInputInfo(error?: React.ReactChild, info?: React.ReactChild) {
  return <InputNotification error={error} info={info} />;
}
```